test/distributed/org/apache/cassandra/distributed/mock/nodetool/NetstatsTest.java

```diff
@@ -0,0 +1,599 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.distributed.mock.nodetool;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.UUID;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Stream;
+
+import org.junit.Assert;
+import org.junit.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.datastax.driver.core.Session;
+import org.apache.cassandra.distributed.Cluster;
+import org.apache.cassandra.distributed.api.IInvokableInstance;
+import org.apache.cassandra.distributed.shared.AbstractBuilder;
+import org.apache.cassandra.distributed.shared.NetworkTopology;
+import org.apache.cassandra.distributed.shared.NodeToolResultWithOutput;
+import org.apache.cassandra.distributed.test.TestBaseImpl;
+import org.apache.cassandra.distributed.util.NodetoolUtils;
+import org.apache.cassandra.service.StorageService;
+import org.apache.cassandra.utils.Pair;
+
+import static java.util.stream.Collectors.toList;
+import static org.apache.cassandra.distributed.api.Feature.GOSSIP;
+import static org.apache.cassandra.distributed.api.Feature.NATIVE_PROTOCOL;
+import static org.apache.cassandra.distributed.api.Feature.NETWORK;
+
+public class NetstatsTest extends TestBaseImpl
+{
+
+    private static final Logger logger = LoggerFactory.getLogger(NetstatsTest.class);
+
+    @Test
+    public void testNetstatsStreamingProgress() throws Exception
+    {
+        final ExecutorService executorService = Executors.newFixedThreadPool(2);
+
+        try (final Cluster cluster = getCluster().start())
+        {
+            final IInvokableInstance node1 = cluster.get(1);
+            final IInvokableInstance node2 = cluster.get(2);
+
+            createTable(cluster);
+
+            node1.flush("netstats_test");
+            node2.flush("netstats_test");
+
+            disableCompaction(node1);
+            disableCompaction(node2);
+
+            populateData();
+
+            //change RF from 1 to 2 so we need to repair it, repairing will causes streaming shown in netstats
+            changeReplicationFactor();
+
+            final Future<NetstatResults> resultsFuture1 = executorService.submit(new NetstatsCallable(node1));
+
+            node1.asyncCallsOnInstance(() -> {
```

Comment:
should replace with

```
node1.nodetoolResult("repair", "netstats_test").asserts().success();
```

This will block the main thread but removes the timeout from the result future if the repair takes longer than 1m.